### PR TITLE
[magnum] Removing quotes around tools list where vcpkg_copy_tools is involked

### LINKED ISSUE #14329: [magnum] build failure - actual issue is vcpkg_copy_tools.cmake
**Host Environment**
- OS: [Microsoft Windows [Version 10.0.19042.610]]
- Compiler: Visual Studio  2019 Version: 16.7.7  VS Tools Version: 14.27.29110
                   Visual Studio  2017 Version: 15.9.28  VS Tools Version: 14.16.27023
- See **Additional Context** below.

**To Reproduce**
`.\vcpkg.exe  install --x-use-aria2 --recurse --keep-going --editable  magnum[al-info,anyaudioimporter,anyimageconverter,anyimageimporter,anysceneconverter,anysceneimporter,audio,cglcontext,core,debugtools,distancefieldconverter,fontconverter,gl,gl-info,glfwapplication,imageconverter,magnumfont,magnumfontconverter,meshtools,objimporter,opengltester,primitives,scenegraph,sdl2application,shaders,text,texturetools,tgaimageconverter,tgaimporter,trade,wavaudioimporter,wglcontext,windowlesswglapplication]:x64-windows magnum-extras[core,ui]:x64-windows magnum-integration[bullet,core,glm,imgui]:x64-windows magnum-plugins[core,ddsimporter,devilimageimporter,drflacaudioimporter,freetypefont,harfbuzzfont,icoimporter,jpegimageconverter,jpegimporter,miniexrimageconverter,openddl,opengeximporter,pngimageconverter,pngimporter,stanfordimporter,stanfordsceneconverter,stbimageconverter,stbimageimporter,stlimporter]:x64-windows `

**Failure logs** 
- Cut and past the appropriate build messages from the console output.
`Computing installation plan...`
`The following packages will be built and installed:`
`    magnum[al-info,anyaudioimporter,anyimageconverter,anyimageimporter,anysceneconverter,anysceneimporter,audio,cglcontext,core,debugtools,distancefieldconverter,fontconverter,gl,gl-info,glfwapplication,imageconverter,magnumfont,magnumfontconverter,meshtools,objimporter,opengltester,primitives,scenegraph,sdl2application,shaders,text,texturetools,tgaimageconverter,tgaimporter,trade,wavaudioimporter,wglcontext,windowlesswglapplication]:x64-windows`
`    magnum-extras[core,ui]:x64-windows`
`    magnum-integration[bullet,core,glm,imgui]:x64-windows`
`    magnum-plugins[core,ddsimporter,devilimageimporter,drflacaudioimporter,freetypefont,harfbuzzfont,icoimporter,jpegimageconverter,jpegimporter,miniexrimageconverter,openddl,opengeximporter,pngimageconverter,pngimporter,stanfordimporter,stanfordsceneconverter,stbimageconverter,stbimageimporter,stlimporter]:x64-windows`
`Starting package 1/4: magnum:x64-windows`
`Building package magnum[al-info,anyaudioimporter,anyimageconverter,anyimageimporter,anysceneconverter,anysceneimporter,audio,cglcontext,core,debugtools,distancefieldconverter,fontconverter,gl,gl-info,glfwapplication,imageconverter,magnumfont,magnumfontconverter,meshtools,objimporter,opengltester,primitives,scenegraph,sdl2application,shaders,text,texturetools,tgaimageconverter,tgaimporter,trade,wavaudioimporter,wglcontext,windowlesswglapplication]:x64-windows...`
`-- Using cached D:/vcpkg_x64/vs2019/prod/downloads/mosra-magnum-v2020.06.tar.gz`
`-- Extracting source D:/vcpkg_x64/vs2019/prod/downloads/mosra-magnum-v2020.06.tar.gz`
`-- Applying patch 001-tools-path.patch`
`-- Applying patch 002-sdl-includes.patch`
`-- Using source at D:/vcpkg_x64/vs2019/prod/buildtrees/magnum/src/v2020.06-97c0869c1f`
`-- Configuring x64-windows`
`-- Building x64-windows-dbg`
`-- Building x64-windows-rel`
`-- magnum-al-infomagnum-distancefieldconvertermagnum-fontconvertermagnum-gl-infomagnum-imageconverter`
**`CMake Error at scripts/cmake/vcpkg_copy_tools.cmake:45 (if):`**
**`  if given arguments:`**
` `
**`    "EXISTS" "D:/vcpkg_x64/vs2019/prod/packages/magnum_x64-windows/bin/magnum-al-info" "magnum-distancefieldconverter" "magnum-fontconverter" "magnum-gl-info" "magnum-imageconverter.exe"`**
` `
**`  Unknown arguments specified`**
`Call Stack (most recent call first):`
`  ports/magnum/portfile.cmake:105 (vcpkg_copy_tools)`
`  scripts/ports.cmake:135 (include)`
` `
` `
`Error: Building package magnum:x64-windows failed with: BUILD_FAILED`
`Please ensure you're using the latest portfiles with `.\vcpkg update`, then`
`submit an issue at https://github.com/Microsoft/vcpkg/issues including:`
`  Package: magnum:x64-windows`
`  Vcpkg version: 2020.06.15-nohash`
` `
`Additionally, attach any relevant sections from the log files above.`
- Please attach any additional failure logs mentioned in the console output.
N/A

**Additional context**
Add any other context about the problem here, such as what you have already tried to resolve the issue.

1. The actual problem appears to be related to the change made to **vcpkg_copy_tools.cmake line 29 PR #13698 [vcpkg] FIx more cases of semicolon mishandling in "scripts" - followup to PR #12926 https://github.com/microsoft/vcpkg/commit/0ff75ac80b9a3770afb3d2971c572628d80a631e#diff-ab246505744d35f525b9356b69b69acb36b60b8f1a99300a2f05bff5d4ab77f2** was changed from:
`cmake_parse_arguments(_vct "AUTO_CLEAN" "SEARCH_DIR" "TOOL_NAMES" ${ARGN})` to
`cmake_parse_arguments(PARSE_ARGV 0 _vct "AUTO_CLEAN" "SEARCH_DIR" "TOOL_NAMES")`
2. Somehow this change causes **vcpkg_copy_tools.cmake line 45 if state to fail: https://github.com/microsoft/vcpkg/blob/e9ff3cd5a04cd0e8122ff56e9873985ff71aa3ca/scripts/cmake/vcpkg_copy_tools.cmake#L45**
3. If I back out this single change to **vcpkg_copy_tools.cmake line 29** only, then magnum and related features and other ports with magnum dependency install normally.

### PR DESCRIPTION
**Pull request by PhoebeHui**
Fixes https://github.com/microsoft/vcpkg/issues/14329

emoving quotes around tools list where vcpkg_copy_tools is involked, see details in https://github.com/microsoft/vcpkg/issues/14329#issuecomment-721027484

### CODE REVIEW

**Comment by jgehw** [comment]
Sorry for not replying earlier (I didn't get notified that I was mentioned in the issue).

This is not the way we should fix it, because removing the backslashes is essentially a work-around to remove work-arounds in the portfiles, which aren't needed any more. Moreover, this could breaks things.

To explain the background, as it's really anything else than obvious:
- cmake uses semicolons as list separators
- => semicolons inside strings that a elements of a list must be escaped
- cmake has two bugs: correctly escaped semicolons get broken, if
  - `cmake_parse_arguments` is used not the `PARSE_ARGV`-way
  - lists are modified other than by appending, i.e. inserting
- That's why I fixed all `cmake_parse_arguments` in #13968
- various portfiles used work-arounds in the past to get around this, in particular, by passing a list as string, for example:
```
set(_TOOL_EXEC_NAMES "")
list(APPEND _TOOL_EXEC_NAMES "some_tool")
list(APPEND _TOOL_EXEC_NAMES "another_tool")
# passing ${_TOOL_EXEC_NAMES} would be passing the list
# passing "${_TOOL_EXEC_NAMES}" converts the list into a string
# so this way to invoke it is a work-around for the buggy cmake API
vcpkg_copy_tools(TOOL_NAMES "${_TOOL_EXEC_NAMES}" AUTO_CLEAN)
# now that the non-buggy version of cmake_parse_arguments is used, it must be invoked without string-conversion work-around
vcpkg_copy_tools(TOOL_NAMES ${_TOOL_EXEC_NAMES} AUTO_CLEAN)
```
- Thus, the way to fix this, is to remove the list-to-string-conversion work-around from the portfile, i.e. drop the quotes around tools list where `vcpkg_copy_tools` is invoked. In #13968 I was hoping to remove all such work-arounds from all portfiles, but apparently, I missed the magnum port.

Long story short: please fix by removing quotes in magnum portfile.

**Comment by PhoebeHui** [comment]
@jgehw, thanks for your review! I have updated the magnum port.

**Review by JackBoosY — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks :D
